lib/Command/SetupCommand.php

```diff
@@ -0,0 +1,200 @@
+ *
+ * @author Matias De lellis <mati86dl@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\FaceRecognition\Command;
+
+use OCP\ITempManager;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use OCA\FaceRecognition\Service\ModelService;
+
+class SetupCommand extends Command {
+
+	/** @var ModelService */
+	protected $modelService;
+
+	/** @var ITempManager */
+	protected $tempManager;
+
+	/** @var OutputInterface */
+	protected $logger;
+
+	/* @var string */
+	protected $tempFolder;
+
+	/*
+	 * Model 1
+	 */
+	private $detectorModelUrl = 'https://github.com/davisking/dlib-models/raw/94cdb1e40b1c29c0bfcaf7355614bfe6da19460e/mmod_human_face_detector.dat.bz2';
+	private $detectorModel = 'mmod_human_face_detector.dat';
+
+	private $resnetModelUrl = 'https://github.com/davisking/dlib-models/raw/2a61575dd45d818271c085ff8cd747613a48f20d/dlib_face_recognition_resnet_model_v1.dat.bz2';
+	private $resnetModel = 'dlib_face_recognition_resnet_model_v1.dat';
+
+	private $predictorModelUrl = 'https://github.com/davisking/dlib-models/raw/4af9b776281dd7d6e2e30d4a2d40458b1e254e40/shape_predictor_5_face_landmarks.dat.bz2';
+	private $predictorModel = 'shape_predictor_5_face_landmarks.dat';
+
+	/**
+	 * @param ModelService $modelService
+	 * @param ITempManager $tempManager
+	 */
+	public function __construct(ModelService $modelService,
+	                            ITempManager $tempManager) {
+		parent::__construct();
+
+		$this->modelService = $modelService;
+		$this->tempManager  = $tempManager;
+	}
+
+	protected function configure() {
+		$this
+			->setName('face:setup')
+			->setDescription('Download and Setup the model 1 used for the analysis');
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+
+		$this->logger = $output;
+
+		$this->modelService->useModelVersion(1);
+		$this->tempFolder = $this->tempManager->getTemporaryFolder('/facerecognition/');
+
+		$this->downloadModel ($this->detectorModelUrl);
```

Comment:
consider to skip downloading if `modelService->modelFilesPresent` is true? Do you want to be sure that files are there?

Reply:
And dont use an magic '1'

Reply:
Done..